### DISCUSSION TITLE
Is it a bug ? My App ID seems to be modified when publishing IPA file...

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#1489] [Q&A] [answered]
I think there is a bug with Adobe AIR when creating the IPA file. 
Everytime I create it, when I want to upload it to the App Store I've got this error : 
"no app found with this signature. Check if "nc.cocogeek.mareeNCd" is correct."

![image](https://user-images.githubusercontent.com/6382334/147304412-da93374f-2525-4432-806f-e8c3eeaeae71.png)

But I've checked, and checked again, I don't understand why there's the "d" at the end. There's not supposed to have a "d". 
See here : 

![image](https://user-images.githubusercontent.com/6382334/147304424-3cfdefa9-b944-48d0-bddc-b89bd78dcfd6.png)

I have a feeling that when publishing my IPA with Adobe Animate it's making a bug and add this "d" at the end of the app id... (I've tried with Adobe Animate 2021 and 2022) and with AIR 33.1.1.713

Is that possible ? 
What can I do ?

**Comment by ajwfrost** [comment, marked as answer]
Hi
Can we take a look at the IPA file that's generated please? You can upload it via the below link:
https://transfer.harman.com/requests/esskAHlne4zG5iRYcrjGdC
thanks

**Reply by ajwfrost** [reply]
Hi - thanks for that.
Not sure where the "d" comes from, but your application descriptor file has:
`<id>nc.cocogeek.mareeNC</id>`
and the resulting Info.plist file in the IPA file has this:
```
        <key>CFBundleIdentifier</key>
        <string>nc.cocogeek.mareeNC</string>
```

The provisioning profile has entitlements:
```
<key>application-identifier</key>
<string>L8N8KB56L6.nc.cocogeek.mareesNC</string>
```

So I am wondering if you updated the application ID in your main app descriptor file to add the 's', it would then match:
```
<id>nc.cocogeek.mareeNC</id>
becomes
<id>nc.cocogeek.mareesNC</id>
```

thanks

**Comment by stephanberger4** [comment]
You’re Indeed correct ! I didn’t think it could be it as without the « s » it was for the Android version.. so I thought the IPA file wouln’t read/care about this section in the manifest.

Anyway, THANK YOU VERY MUCH !!!!!!

Merry Christmas,

Best regards,

Stephan BERGER

________________________________
De : Andrew Frost ***@***.***>
Envoyé : Friday, December 24, 2021 4:43:10 PM
À : airsdk/Adobe-Runtime-Support ***@***.***>
Cc : stephanberger4 ***@***.***>; Author ***@***.***>
Objet : Re: [airsdk/Adobe-Runtime-Support] Is it a bug ? My App ID seems to be modified when publishing IPA file... (Discussion #1489)


Hi - thanks for that.
Not sure where the "d" comes from, but your application descriptor file has:
<id>nc.cocogeek.mareeNC</id>
and the resulting Info.plist file in the IPA file has this:

        <key>CFBundleIdentifier</key>
        <string>nc.cocogeek.mareeNC</string>


The provisioning profile has entitlements:

<key>application-identifier</key>
<string>L8N8KB56L6.nc.cocogeek.mareesNC</string>


So I am wondering if you updated the application ID in your main app descriptor file to add the 's', it would then match:

<id>nc.cocogeek.mareeNC</id>
becomes
<id>nc.cocogeek.mareesNC</id>


thanks

—
Reply to this email directly, view it on GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fairsdk%2FAdobe-Runtime-Support%2Fdiscussions%2F1489%23discussioncomment-1866393&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=SgqgKZve3c5CNv4vNm%2FB2ILVa%2FtPahTEp7Om87ALqqk%3D&reserved=0>, or unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABQWF7UDX2T3JPUPDL3RCF3USQB65ANCNFSM5KVZHUYQ&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=HuPWifEtNObYHt%2B%2BzJ8BFd0nQWgm2bR7VVG9Yr8rNYo%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=380zRZxBtDGJeCeP4X0hAaDhkQ9D3PyMCSRrd4lEiZw%3D&reserved=0> or Android<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2FUKaHnDKazeVdC0Iao9wKLm2r%2FajkQPztqCemTSMrlw%3D&reserved=0>.

**Comment by stephanberger4** [comment]
Hmm I’ve maybe talked too fast. Here’s the new error :

![image](https://user-images.githubusercontent.com/6382334/147325152-2bb88748-17d8-4d54-800b-77469663c35f.png)


I’ll investigate on it.

Regards

Stephan BERGER

De : Stéphan ***@***.***>
Envoyé le :vendredi 24 décembre 2021 16:53
À : ***@***.***>
Objet :RE : [airsdk/Adobe-Runtime-Support] Is it a bug ? My App ID seems to be modified when publishing IPA file... (Discussion #1489)

You’re Indeed correct ! I didn’t think it could be it as without the « s » it was for the Android version.. so I thought the IPA file wouln’t read/care about this section in the manifest.

Anyway, THANK YOU VERY MUCH !!!!!!

Merry Christmas,

Best regards,

Stephan BERGER


De : Andrew Frost ***@***.***>
Envoyé : Friday, December 24, 2021 4:43:10 PM
À : airsdk/Adobe-Runtime-Support ***@***.***>
Cc : stephanberger4 ***@***.***>; Author ***@***.***>
Objet : Re: [airsdk/Adobe-Runtime-Support] Is it a bug ? My App ID seems to be modified when publishing IPA file... (Discussion #1489)


Hi - thanks for that.
Not sure where the "d" comes from, but your application descriptor file has:
<id>nc.cocogeek.mareeNC</id>
and the resulting Info.plist file in the IPA file has this:

        <key>CFBundleIdentifier</key>

        <string>nc.cocogeek.mareeNC</string>

The provisioning profile has entitlements:

<key>application-identifier</key>

<string>L8N8KB56L6.nc.cocogeek.mareesNC</string>

So I am wondering if you updated the application ID in your main app descriptor file to add the 's', it would then match:

<id>nc.cocogeek.mareeNC</id>

becomes

<id>nc.cocogeek.mareesNC</id>

thanks

—
Reply to this email directly, view it on GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fairsdk%2FAdobe-Runtime-Support%2Fdiscussions%2F1489%23discussioncomment-1866393&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=SgqgKZve3c5CNv4vNm%2FB2ILVa%2FtPahTEp7Om87ALqqk%3D&reserved=0>, or unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABQWF7UDX2T3JPUPDL3RCF3USQB65ANCNFSM5KVZHUYQ&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=HuPWifEtNObYHt%2B%2BzJ8BFd0nQWgm2bR7VVG9Yr8rNYo%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=380zRZxBtDGJeCeP4X0hAaDhkQ9D3PyMCSRrd4lEiZw%3D&reserved=0> or Android<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7C1492574c5c394269512008d9c6a04546%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637759213920907922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2FUKaHnDKazeVdC0Iao9wKLm2r%2FajkQPztqCemTSMrlw%3D&reserved=0>.

**Reply by ajwfrost** [reply]
I don't think the new error details made it up into Github via the email response...?

**Reply by stephanberger4** [reply]
Hmm I’ve maybe talked too fast. Here’s the new error : 

![image](https://user-images.githubusercontent.com/6382334/147325075-b9b4045b-c9ac-4844-a4d7-c15eab1ddcfd.png)

I’ll investigate on it.

**Reply by ajwfrost** [reply]
Yes your app descriptor has this key in it, which is being put into the entitlements embedded into the code signature:
```
<key>aps-environment</key>
<string>production</string>
```
which is related to push notifications. You perhaps need to select that on the provisioning profile too, assuming you're using it (or if not, just delete that entitlement from the app descriptor)

thanks

**Comment by stephanberger4** [comment]
Oh... I'll check that !

Thank you again for everything !

Stephan Berger
________________________________
